carma_wm/include/carma_wm/SignalizedIntersectionManager.hpp

```diff
@@ -144,6 +155,69 @@ class SignalizedIntersectionManager
   */
   lanelet::Lanelets identifyInteriorLanelets(const lanelet::Lanelets& entry_llts, const std::shared_ptr<lanelet::LaneletMap>& map);
 
+  /**
+   * @brief processSpatFromMsg update map's traffic light states with SPAT msg
+   *
+   * @param spat_msg Msg to update with
+   */
+  void processSpatFromMsg(const carma_v2x_msgs::msg::SPAT& spat_msg,
+    const std::shared_ptr<lanelet::LaneletMap>& semantic_map);
+
+  /*!
+  *  \brief Update the recorded traffic signal objects in the intersection as fixed signal,
+  *         determining the phase cycles from the available signal durations.
+  *         Do nothing if intersection_id is not found.
+  *  \param intersection_id of the intersection
+  *  \param lanelet_map to update the regulatory elements according to internal signals recorded
+  */
+  void updateSignalAsFixedSignal(uint16_t intersection_id,
+    const std::shared_ptr<lanelet::LaneletMap>& semantic_map);
+
+  /*!
+  *  \brief Update the recorded traffic signal objects in the intersection as dynamic signal,
+  *         directly overwriting any previous states
+  *  \param intersection_id of the intersection
+  *  \param lanelet_map to update the regulatory elements according to internal signals recorded
+  */
+  void updateSignalAsDynamicSignal(uint16_t intersection_id,
+    const std::shared_ptr<lanelet::LaneletMap>& semantic_map);
+
+  /*!
+  *  \brief Extract tuple of 1) vector of pair of traffic signal states' min_end_time and their
```

Comment:
Is there any benefit to extracting these variables as a tuple with min_end_time and states and then a separate vector for start time? Could we use a map from state to a tuple of start_time and min_end_time?